controllers/platform/k8s.go

```diff
@@ -61,14 +62,14 @@ func (action *serviceAction) Handle(ctx context.Context, platform *operatorapi.S
 		return nil, err
 	}
 
-	psDI := services.NewDataIndexHandler(platform)
+	psDI := services.NewDataIndexHandler(ctx, action.client, platform)
 	if psDI.IsServiceSetInSpec() {
 		if err := createOrUpdateServiceComponents(ctx, action.client, platform, psDI); err != nil {
 			return nil, err
 		}
 	}
 
-	psJS := services.NewJobServiceHandler(platform)
+	psJS := services.NewJobServiceHandler(ctx, action.client, platform)
```

Comment:
There's no need to pass the `client` reference anymore. We can access it from the `utils` package.